ServerCore/PuzzleStateHelper.cs

```diff
@@ -569,6 +569,14 @@ public static async Task UpdateTeamsWhoSentResponse(PuzzleServerContext context,
 
                     await context.SaveChangesAsync();
                     transaction.Commit();
+
+                    var teamMembers = await (from TeamMembers tm in context.TeamMembers
```

Comment:
Thanks for doing this. I wrote this but hadn't had time to verify it. This is similar to what I wrote except that we already know the relevant teams. Here's what I wroteto collect the primary emails:
```
List<String> primaryContacts = (from matchingResponse in submissionsThatMatchResponse
                               select matchingResponse.State.Team.PrimaryContactEmail).ToList();
```
and here's the replacement untested code when you said to send to all team members:
```
HashSet<int> TeamIDs = new HashSet<int>(from matchingResponse in submissionsThatMatchResponse
                                        select matchingResponse.State.Team.ID);

List<String> teamMembers = await (from TeamMembers tm in context.TeamMembers
                                  where TeamIDs.Contains(tm.Team.ID)
                                  select tm.Member.Email).ToListAsync();
```

Reply:
How on earth did I miss that?!?! Thanks, I feel kind of silly right now. :-)

Reply:
I'm not certain, but it's possible the HashSet version could end up pulling all TeamMembers from the database and then checking them in the frontend's HashSet. If performance is an issue on this page, we might need to measure it.

Reply:
After looking at the code more carefully: I do think it is important to send to all players rather than primary contacts because most teams do not have a group email and you never know who's solving what. The old site did it this way. But I will be careful thinking about what data I might already have. And Morgan's statement concerns me enough that I'm not sure which version I'll use. But I'll definitely pick one of them :-)

Reply:
I'm opting to leave this as is for now but I am keeping Bruce's alternative in mind. I'll try to measure this when I can, by changing a response in the sandbox after it's deployed and configured.